src/OrchardCore.Modules/OrchardCore.Lucene/LuceneIndexManager.cs

```diff
@@ -75,7 +75,7 @@ public void DeleteDocuments(string indexName, IEnumerable<string> contentItemIds
                 {
                     pool.MakeDirty();
                 }
-            });
+            }, true);
```

Comment:
This is closing the index after each operation. That solves the issue but impacts the perf a lot.
I assume you are using the same filesystem for two different instances. In this case this is the issue, Lucene doesn't support that, we need one index folder per instance. We need to thing about it more.

Reply:
Yes exactly, i will think about your suggestion.